src/gallium/drivers/lima/lima_resource.c

```diff
@@ -82,21 +111,13 @@ lima_resource_create(struct pipe_screen *pscreen,
    if (!res->bo)
       goto err_out0;
 
-   if (screen->ro && templat->bind & PIPE_BIND_SCANOUT) {
-      res->scanout =
-         renderonly_scanout_for_resource(pres, screen->ro, NULL);
-      if (!res->scanout)
-         goto err_out1;
-   }
-
+ok_out:
```

Comment:
I'd recommend you put renderonly and normal resource create path in two functions instead of mix gotos